src/form-builder.tsx

```diff
@@ -34,22 +68,26 @@ export const FormBuilder = () => {
     // =========================================================================
 
     return (
-        <BuilderProvider>
-            <DisplayProvider>
-                <Wrapper>
-                    {!isLargeScreen && <ScreenNotSupportedErrorDisplay />}
-                    <Container
-                        type="grid"
-                        stretch
-                        $isLargeScreen={isLargeScreen}
-                    >
-                        <Toasts />
-                        <Modals />
-                        <MainPanel />
-                        <SidePanel />
-                    </Container>
-                </Wrapper>
-            </DisplayProvider>
-        </BuilderProvider>
+        <Wrapper>
+            {!isLargeScreen && <ScreenNotSupportedErrorDisplay />}
+            <Container type="grid" stretch $isLargeScreen={isLargeScreen}>
+                <Toasts />
+                <Modals />
+                <MainPanel />
+                <SidePanel offset={offset} />
+            </Container>
+        </Wrapper>
     );
-};
+});
+
+export const FormBuilder = forwardRef<IFormBuilderMethods, IProps>(
+    ({ offset }, ref) => {
+        return (
+            <BuilderProvider>
+                <DisplayProvider>
+                    <Component ref={ref} offset={offset} />
+                </DisplayProvider>
+            </BuilderProvider>
+        );
+    }
+);
```

Comment:
there is no point destructuring the props prematurely - they arent use here and new props have to be added manually

```suggestion
export const FormBuilder = forwardRef<IFormBuilderMethods, IProps>(
    (props, ref) => {
        return (
            <BuilderProvider>
                <DisplayProvider>
                    <Component ref={ref} {...props} />
                </DisplayProvider>
            </BuilderProvider>
        );
    }
);
```